include/drivers/ieee802154/rf2xx.h

```diff
@@ -132,4 +132,17 @@ struct rf2xx_context {
 	u8_t rx_phr;
 };
 
-#endif /* ZEPHYR_DRIVERS_IEEE802154_IEEE802154_RF2XX_H_ */
+/**
+ * @brief Set mac address to be used on transceiver
+ *
+ * The driver by default uses local_mac_address from Device Tree with
+ * fallback to random mac. This is a public method that allows override,
+ * since internally is defined as a weak method.
+ *
+ * @param[in] dev   Transceiver device instance
+ *
+ * @return pointer to mac_addr variable defined on the driver context
+ */
+u8_t *rf2xx_get_mac(struct device *dev);
```

Comment:
Nack on that one. 
First because it is not generic among all 15.4 drivers. But also because it is useless: you just need to set the extended address to the value you want and that's it and there is already a generic way to set it by the user.

Reply:
I understood the reason but I can't find a good solution. The issue #23193 explain that MAC must be set "before" the driver init. This is because RF2xx is a external (not built in on SoC) radio. On real application the EUI-64 will be read from some memory and applied on the transceiver. This step is only possible when app already started. For instance, OpenThread si already started when user can run NET_REQUEST_IEEE802154_SET_EXT_ADDR. This could let driver send packages with different MACs. I'm assume this behavior could block a certification process, not sure about it. Kconfig and DT won't work here and only are valid for tests purposes.

That said, I agreed this is not generic among all 15.4 drivers. I agree in part on the second argument about be useless, it's a diffente way to get same goal :D. Anyway, I was wondering if driver could be blocked until it receives some API commands like NET_REQUEST_IEEE802154_SET_EXT_ADDR (and others maybe). This way, driver can be enabled but will be fully operational only when receives such commands.. It could not be like others but may should solve the problem without harm to much.

Is acceptable block driver that way? Any options about it? Implications on OT and/or Zigbee?

Reply:
It looks like there is a simple way to prevent the driver from sending packets with wrong MAC. Just call `net_if_flag_set(iface, NET_IF_NO_AUTO_START);` at the end of `rf2xx_iface_init()`, that will prevent the network interface from coming up automatically. Then just call `net_if_up()` after receiving `NET_REQUEST_IEEE802154_SET_EXT_ADDR`.